tests/Purl/Test/UrlTest.php

```diff
@@ -362,6 +362,10 @@ public function testRelativeUrl() : void
         $url = new Url('/events');
         $url->query->set('param1', 'value1');
         $this->assertEquals('/events?param1=value1', (string) $url);
+
+        // test fragment with colon
+        $url = new Url('http://example.com/#hello:123');
+        $this->assertEquals('http://example.com/', (string) $url);
```

Comment:
@jwage, yes. It's fixed that issue. Please see more details about that.
It have removed the hash bang.

Reply:
This doesn't feel right to me. It shouldn't just silently remove it from the URL.

Reply:
What do you think about this improvement?
I think we can fetch this hash bang and store another key in ```Url``` instance.

What do you think?

Reply:
Are hashbangs technically allowed to have a colon in them?

Reply:
I don't think that hash bang with colon is not the standard.

Reply:
Then I am not sure if there is anything to fix here. Dropping the hashbang value entirely is not a good fix. I would maybe accept an enhancement to allow the hashbang with the colon like you described but that would be a more involved fix.

Reply:
Consider the approach:

- parse the url.
- get the fragment.
- validate the fragment whether it's with the colon.

What do you think? Thanks.

Reply:
We can give it a try.